turn/turn_rest.go

```diff
@@ -0,0 +1,58 @@
+package turn
+
+import (
+	"crypto/hmac"
+	"crypto/sha1"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/rs/xid"
+	"github.com/screego/server/config"
+)
+
+type TurnREST struct {
+	ttl    time.Duration
+	secret []byte
+	port   int
+}
+
+func (t *TurnREST) AcceptAccounts(client, host *TurnAccount) error {
+	err := buildPassword(client, t.ttl, t.secret)
+	if err != nil {
+		return err
+	}
+	return buildPassword(host, t.ttl, t.secret)
+}
+
+func buildPassword(account *TurnAccount, ttl time.Duration, secret []byte) error {
+	//https://stackoverflow.com/questions/35766382/coturn-how-to-use-turn-rest-api#54725092
+	if ttl <= 0 {
+		return errors.New("Use a TTL > 0")
+	}
+	account.Username = fmt.Sprintf("%d:%s", time.Now().Add(ttl).Unix(), account.Id.String())
+	mac := hmac.New(sha1.New, secret)
+	_, err := mac.Write([]byte(account.Username))
+	if err != nil {
+		return err
+	}
+	account.Credential = string(mac.Sum(nil))
```

Comment:
This needs to be base64-encoded, i.e. 

```go
account.Credential = base64.StdEncoding.EncodeToString(mac.Sum(nil))
``` 

(and import `"encoding/base64"`)

Reply:
See 1ebe9ab6cd972fa4d4c45091d35d25ab86185990